crypto/err/err_locl.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 1995-2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#define err_allocate(p)                                         \
```

Comment:
these are not time-critical, just make them internal functions.  Then you don't have to worry about parenthesizing or someone calling ```err_clear_data(p, i++)```

Reply:
```err_allocate``` is a misleading name since it doesn't actually allocate anything.

Reply:
But it does allocate a new error record slot. Sure, it's in a ring buffer, but still. Allocation as a term covers more than merely allocating memory on the heap, doesn't it? I'm pretty sure I have seen that word used more broadly.

However, feel free to suggest a name that you find more suitable.

Reply:
```err_get_slot``` ?

Reply:
Sure

Reply:
Is the name changing for this one?

I tend to agree that (internal) functions might be a nicer choice.